### PR TITLE
Modifies wrong path for startup.py initialization file.

### LINKED ISSUE #3722: startup.py path in documentation not the same than in QGIS 3.6
## Description

Page URL: https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/intro.html?highlight=startup#id7

the correct path in linux, is .local/share/QGIS/QGIS3

I had malfunction on startup and asked in gis.stackexchane : https://gis.stackexchange.com/questions/322007/qgis-3-6-startup-py-not-executed?noredirect=1#comment524036_322007

I do not know how to touch the docs directly and make a PR.

Thankyou.

### PR DESCRIPTION
**Pull request by JaumeFigueras**
### Description

Goal: Solved wrong path for startup.py initialization file.

Ticket: fix #3722 

- [X] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [X] The description of this PR is coherent with the manual and does not provide wrong information.
- [x] This PR passes the checks.

### CODE REVIEW

**Comment by DelazJ** [comment]
This looks great @JaumeFigueras

File: source/docs/pyqgis_developer_cookbook/intro.rst [hunk 1]
```diff
@@ -127,9 +127,9 @@ The :file:`startup.py` file
 
 Every time QGIS starts, the user's Python home directory
```
Comment (DelazJ) [inline source/docs/pyqgis_developer_cookbook/intro.rst]: Anyone knows how we can refer to the QGIS3 folder? It's not the `user's Python home directory`. The user's application directory? Anything better?

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/intro.rst]: `the user's QGIS directory`?

Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/intro.rst]: I prefer mine, especially since there's a QGIS folder in the path...

Comment (DelazJ) [reply source/docs/pyqgis_developer_cookbook/intro.rst]: Not a single third voice to help take a decision? 😞 
Merging as is.... (with the inappropriate introduction)

**Comment by DelazJ** [comment]
Thanks @JaumeFigueras

**Pull request merged**